### PR TITLE
[url_launcher] Use `url_launcher_platform_interface` to handle calls

### PR DESCRIPTION
**Pull request by hterkelsen**
## Description

Forward calls from `package:url_launcher` to the platform-specific implementation declared in `UrlLauncherPlatform.instance`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/url_launcher/url_launcher/test/url_launcher_test.dart [hunk 1]
```diff
@@ -56,142 +49,128 @@ void main() {
...
     await closeWebView();
-    expect(
-      log,
-      <Matcher>[isMethodCall('closeWebView', arguments: null)],
-    );
+    verify(mock.closeWebView());
   });
 }
+
+class MockUrlLauncher extends Mock implements UrlLauncherPlatform {}
```
Comment (amirh) [inline packages/url_launcher/url_launcher/test/url_launcher_test.dart]: This was a red flag for me as I was planning to enforce that it's not implemented with `implements`.
I sent #2230 which leaves a back door for mocks.

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 2]
```diff
@@ -14,10 +14,13 @@ flutter:
 dependencies:
   flutter:
     sdk: flutter
+  url_launcher_platform_interface: ^1.0.0
```
Comment (amirh) [inline packages/url_launcher/url_launcher/pubspec.yaml]: If this lands after #2230 lets make this ^1.01

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 3]
```diff
@@ -14,10 +14,13 @@ flutter:
...
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
+  test: ^1.3.0
+  mockito: ^3.0.0
```
[the remark this answers is not in the document]
Comment (hterkelsen) [reply packages/url_launcher/url_launcher/pubspec.yaml]: No strong preference, just saw that `^3.0.0` was used in other plugins in this repo.

[the remark this answers is not in the document]
Comment (hterkelsen) [reply packages/url_launcher/url_launcher/pubspec.yaml]: Switched to ^4.1.1

File: packages/url_launcher/url_launcher/pubspec.yaml (hunk 2 above)
Comment (hterkelsen) [reply packages/url_launcher/url_launcher/pubspec.yaml]: Done.

File: packages/url_launcher/url_launcher/test/url_launcher_test.dart (hunk 1 above)
Comment (hterkelsen) [reply packages/url_launcher/url_launcher/test/url_launcher_test.dart]: Done.

**Pull request merged**